### PR TITLE
PR: Add Contributing file

### LINKED ISSUE #3573: Create requirements file to ease introduction to development and testing
http://conda.pydata.org/docs/using/envs.html

### LINKED ISSUE #3986: Write additional information on 3.1, 3.x and master branches and how to do bugfixes
Use the wiki to include information on the current push process

### PR DESCRIPTION
**Pull request by rlaverde**
Fixes #3986 
Fixes #3573

The issue is label for 3.2 but I made it against 3.1.x because It doesn't change anything (only add documentation), and could me merged early.

When creating requirements file I noticed that `flaky` isn't available through conda so I make a little exception for this package

### CODE REVIEW

**Comment by goanpeca** [comment]
>When creating requirements file I noticed that flaky isn't available through conda so I make a little exception for this package

What do you mean @rlaverde?

**Comment by rlaverde** [comment]
> What do you mean @rlaverde?

I added this to the Contributing file test dependencies section, and exclude flaky from `test_requirements.txt
`
```bash
  $ conda install --file requirements/requirements.txt && pip install flaky
```

```bash
  $ pip install -r requirements/requirements.txt && pip install flaky
```

**Comment by goanpeca** [comment]
Then please hold this PR until we have the flaky package on the spyder-ide channel. I will make it

**Comment by ccordoba12** [comment]
@rlaverde, please rebase this one in top of `3.x`. This was supposedly simple, but now it requires more work.

**Comment by goanpeca** [comment]
@rlaverde could you update the code to include flaky? the conda-forge package recipe is merged now and the package should land in spyder-ide soon.

**Comment by rlaverde** [comment]
> could you update the code to include flaky? the conda-forge package recipe is merged now and the package should land in spyder-ide soon.

@goanpeca  Done, I also add `spyder-ide` channel to the instructions

@ccordoba12 I rebased it onto 3.x

File: README.md [hunk 1]
```diff
@@ -169,6 +169,9 @@ a Python version greater than 2.7 (Python 3.2 is not supported anymore).
 * **SciPy**: Import Matlab workspace files in the Variable Explorer.
 * **Cython**: Run Cython files in the IPython console.
 
+## Contributing
+
+Everyone is invited to contribute. [Read the contributing instructions](CONTRIBUTING.md), then get started.
```
Comment (rlaverde) [inline README.md]: should I change this for the complete link? 
`https://github.com/spyder-ide/spyder/blob/master/CONTRIBUTING.md`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+### Cloning the repo
+
+```bash
+  $ git clone https://github.com/spyder-ide/spyder.git
+```
+
+### Creating a conda environment (or virtualenv)
+
+```bash
+  $ conda create -n spyder spyder
```
Comment (rlaverde) [inline CONTRIBUTING.md]: should I leave this instruction? or only advice to install dependencies from requirements file?

File: README.md (hunk 1 above)
Comment (ccordoba12) [inline README.md]: Yes, please.

**Comment by goanpeca** [comment]
@rlaverde flaky is now on the anaconda.org/spyder-ide channel

**Pull request merged**